src/ralph/data_center/subscribers.py

```diff
@@ -141,18 +133,54 @@ def handle_update_vip_event(data):
         logger.error(msg, '; '.join(errors))
         return
 
-    ip,  = IPAddress.objects.get_or_create(address=data['ip'])
+    ip, _ = IPAddress.objects.get_or_create(address=data['ip'])
     protocol = VIPProtocol.from_name(data['protocol'].upper())
-    vip = get_vip(ip.address, data['port'], protocol.name)
+    vip = get_vip(ip, data['port'], protocol)
     if vip is None:
+        # VIP not found, should create new one.
+        return handle_create_vip_event(data)
+
+    # update cluster.
+    cluster_type, _ = ClusterType.objects.get_or_create(
```

Comment:
Not sure if I like the idea of creating cluster types this way. Would it not be better to log the fact that a given cluster type does not exist and then create it if required?